airlock/file_browser_api.py

```diff
@@ -227,7 +229,15 @@ def get_workspace_tree(workspace, selected_path=ROOT_PATH, selected_only=False):
 
     else:
         # listing all files in one go is much faster than walking the tree
-        pathlist = [p.relative_to(root) for p in root.glob("**/*")]
+        # Note we limit the files to valid LEVEL4_FILE_TYPES here, so as to
+        # reduce some unnecessary iteration in get_path_tree. However, get_path_tree
+        # ALSO needs to check each file-type node it finds as it builds the tree for
+        # each path in the pathlist
+        pathlist = [
+            p.relative_to(root)
+            for p in root.glob("**/*")
+            if p.is_dir() or p.suffix in LEVEL4_FILE_TYPES
```

Comment:
Nice.

I'm wondering - should we also exclude files starting with `.` here too?